pandas/tests/extension/base/constructors.py

```diff
@@ -33,6 +33,31 @@ def test_series_constructor(self, data):
         assert result2.dtype == data.dtype
         assert isinstance(result2._mgr.blocks[0], ExtensionBlock)
 
+    def test_series_constructor_no_data_with_index(self, dtype, na_value):
+        result = pd.Series(index=[1, 2, 3], dtype=dtype)
+        expected = pd.Series([na_value] * 3, index=[1, 2, 3], dtype=dtype)
+        self.assert_series_equal(result, expected)
+
+        # GH 33559 - empty index
+        result = pd.Series(index=[], dtype=dtype)
+        expected = pd.Series([], index=pd.Index([], dtype="object"), dtype=dtype)
+        self.assert_series_equal(result, expected)
```

Comment:
this PR appears to also close #33559 

the index discrepancy is consistent with non-EA types.

```
>>> import pandas as pd
>>> pd.__version__
'1.1.0.dev0+1446.g1c88e6aff'
>>> pd.Series(dtype="int64", index=[]).index
Index([], dtype='object')
>>>
>>> pd.Series(dtype="int64").index
Index([], dtype='object')
>>>
>>> pd.Series([], dtype="int64").index
RangeIndex(start=0, stop=0, step=1)
>>>
```

Reply:
Yes, there is another PR trying to clean this up